### PR TITLE
dinitcheck: account for service working directory for commands

### PR DESCRIPTION
**Pull request by q66**

### CODE REVIEW

File: src/dinitcheck.cc [hunk 1]
```diff
@@ -429,9 +432,19 @@ service_record *load_service(service_set_t &services, const std::string &name,
 
     settings.finalise(report_err, renvmap, report_err, resolve_var);
 
+    if (!settings.working_dir.empty()) {
+        service_wdir = settings.working_dir;
+    }
+    dirfd = open(service_wdir.c_str(), O_DIRECTORY | O_PATH);
```
Comment (mobin-2008) [inline src/dinitcheck.cc:438]: Hi.

The problem with this PR is that `O_PATH` is not defined by POSIX and it means there is no guarantee about its availability (e.g MacOS ci is failing due to this).

I guess opening with `O_RDONLY` is better option but It makes into problem when directory read access isn't permitted but execute access is permitted.

Ref: https://pubs.opengroup.org/onlinepubs/9699919799/

**Review by mobin-2008 — CHANGES_REQUESTED** [review]

File: src/dinitcheck.cc (hunk 1 above)
Comment (mobin-2008) [reply src/dinitcheck.cc]: I'm not sure opening directory with just `O_DIRECTORY` is safe. Is it?

Comment (q66) [reply src/dinitcheck.cc]: no idea what you mean?

Comment (mobin-2008) [reply src/dinitcheck.cc]: Nevermind, I didn't see O_RDONLY

**Review by davmac314 — APPROVED** [review]
LGTM

**Pull request merged**